enterprise/cypher/spec-suite-tools/src/main/scala/cypher/feature/parser/SideEffects.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package cypher.feature.parser
+
+import cucumber.api.DataTable
+import org.neo4j.graphdb.{Entity, GraphDatabaseService}
+import org.opencypher.tools.tck.InvalidFeatureFormatException
+import org.opencypher.tools.tck.constants.TCKSideEffects._
+
+import scala.collection.JavaConverters._
+
+object SideEffects {
+
+  def expect(expectations: DataTable): Values = {
+    val keys = expectations.transpose().topCells().asScala
+    val values = expectations.transpose().cells(1).asScala.head
+
+    val expected = keys.zipWithIndex.foldLeft(Values()) {
+      case (v, (ADDED_NODES, index)) =>
+        v.copy(nodesCreated = values.get(index).toInt)
+      case (v, (DELETED_NODES, index)) =>
+        v.copy(nodesDeleted = values.get(index).toInt)
+      case (v, (ADDED_RELATIONSHIPS, index)) =>
+        v.copy(relsCreated = values.get(index).toInt)
+      case (v, (DELETED_RELATIONSHIPS, index)) =>
+        v.copy(relsDeleted = values.get(index).toInt)
+      case (v, (ADDED_LABELS, index)) =>
+        v.copy(labelsCreated = values.get(index).toInt)
+      case (v, (DELETED_LABELS, index)) =>
+        v.copy(labelsDeleted = values.get(index).toInt)
+      case (v, (ADDED_PROPERTIES, index)) =>
+        v.copy(propsCreated = values.get(index).toInt)
+      case (v, (DELETED_PROPERTIES, index)) =>
+        v.copy(propsDeleted = values.get(index).toInt)
+      case (sideEffect, _) => throw InvalidFeatureFormatException(
```

Comment:
Did you mean to report on the accumulator here? Seems to make more sense to `case (_, sideEffect)` or `case (_, (sideEffect, _))`